### ISSUE TITLE
Throttle Transcode doesnt work

### ISSUE THREAD

**Issue by RandomDelta6** [jellyfin/jellyfin#2919]
**Describe the bug**
Throttle transcode isn't working. Transcoding stops only after the entire media has been transcoded. [Orange bar way ahead of blue].Throttle transcode box is checked and am using quicksync acceleration.

**System (please complete the following information):**
 - OS: Windows 10
 - Virtualization:no
 - Clients: android 9
 - Browser: [e.g. Firefox 72, Chrome 80, Safari 13]
 - Jellyfin Version: 10.5.4 stable
 - Playback:Transcode
 - Installed Plugins:none
 - Reverse Proxy: none
 - Base URL:none
 - Networking: none
 - Storage: local

**To Reproduce**
playback any hevc video on web browser like chrome

**Expected behavior**
transcode should stop after a little while,instead it is continuing putting unwanted strain on system

**Logs**
n/a

**Screenshots**
![Annotation 2020-04-19 171921](https://user-images.githubusercontent.com/53912269/79686999-ee51d380-8261-11ea-88c9-4ba10602b283.png)

**Comment by Artiume** [comment]
The throttler is only with Software Transcoding, not Hardware Transcoding.

**Comment by RandomDelta6** [comment]
it used to work fine in version 10.4

**Comment by Artiume** [comment]
It didn't exist in 10.4. Please see https://github.com/jellyfin/jellyfin/pull/2258 and it is specifically disabled for HWA.

**Comment by RandomDelta6** [comment]
ok. Thanks. 
Maybe consider enabling throttle transcode for hwa playback?

**Comment by Artiume** [comment]
It was disabled because it caused abnormal behavior. The throttler now is a hack and needs to be reworked. It will probably stay the way it is for a while, it is not a high priority feature.

**Comment by RandomDelta6** [comment]
ok. Thanks!

**Comment by RandomDelta6** [comment]
Since you mentioned it caused it caused abnormal behaviour, I think the code to enable it for hwa exists. Can you tell how to enable it. 
Also, how about giving a new checkbox for hwa throttle transcodes and marking it as experimental?

**Comment by Artiume** [comment]
To enable it for HWA, you'll need to build Jellyfin yourself. 

https://github.com/jellyfin/jellyfin/blob/16401ec7ae078b2b7c4c65e3792cb4c4159490d2/MediaBrowser.Api/Playback/BaseStreamingService.cs#L324

In this section, remove the If statement so it will always return true.

**Comment by bingzhangdai** [comment]
> The throttler now is a hack and needs to be reworked. It will probably stay the way it is for a while, it is not a high priority feature.

From my understanding, this feature will not be available for a long period of time?

**Comment by RandomDelta6** [comment]
just commenting to remove the stale label. Throttle transcode still does not work with Gpu acceleration.
Would be great if it could be added as an experimental toggle

**Comment by bingzhangdai** [comment]
just comment to keep this issue open

**Comment by bingzhangdai** [comment]
just comment to keep this issue open

**Comment by tinect** [comment]
Sadly, this feature still doesn't work at all :-(

**Comment by ragingsonar** [comment]
I really hope this issue is addressed. I can't have jellyfin writing 20gb+ files when the user only watches 10 minutes of a 2 hour video

**Comment by RandomDelta6** [comment]
I agree. It uselessly puts strain on the system and especially wears down the ssd. It's a huge waste in terms of power and drive life, IK it works with Software Transcoding but I require HW transcoding as my system won't be able to handle the streams I require otherwise.

**Comment by ragingsonar** [comment]
> I agree. It uselessly puts strain on the system and especially wears down the ssd. It's a huge waste in terms of power and drive life, IK it works with Software Transcoding but I require HW transcoding as my system won't be able to handle the streams I require otherwise.

I'm pretty sure software transcoding limit is broken now too

**Comment by cvium** [comment]
It's unfortunately a very broken feature. To have proper support we'd need to make changes to ffmpeg itself.

**Comment by ragingsonar** [comment]
> It's unfortunately a very broken feature (and ill conceived). To have proper support we'd need to make changes to ffmpeg itself.

I haven't any experience with this sort of program, so I apologise if this is naive or stupid, but could some sort of cpu limiting solution like https://github.com/opsengine/cpulimit work for when jellyfin launches ffmpeg? As in to implement some sort of non-ideal workaround which limits usage without modifying the programs source?

**Comment by Artiume** [comment]
It would have to be a cross platform solution, ideally from within ffmpeg itself. adding additional processes isn't the best manner for long-term solutions.

**Comment by vgarleanu** [comment]
> It would have to be a cross platform solution, ideally from within ffmpeg itself. adding additional processes isn't the best manner for long-term solutions.

cpulimit works by using SIGSTOP and SIGCONT to pause and unpause a process. These two signals are available on all unix platforms. As for Windows there are the `NtSuspendProcess` and `NtResumeProcess` APIs which have the same effect.

Im sure throttling could be implemented with this.

**Comment by barolo** [comment]
This needs to be adressed. I keep transcode cache in 8GB of RAM, gets fillled up on occasion and has to be purged manually. Otherwise nothing that needs remuxing or transcoding will play.

**Comment by cvium** [comment]
Need is a strong word. It's nice to have, but without proper ffmpeg throttle support it's just not feasible.

**Comment by shaino** [comment]
Recent ffmpeg development updates quickly. Unfortunately, the 'Throttle Transcodes' option is not highly prioritized.
Eager to start as soon as possible.

**Comment by felix920506** [comment]
I have just tested this on my Jellyfin instance (10.8.11, jellyfin-ffmpeg6)
This feature is working so it is likely it was fixed but not closed.

ffmpeg logs:
[throddle.txt](https://github.com/jellyfin/jellyfin/files/12841617/throddle.txt)

**Comment by Shadowghost** [comment]
The fix was backported.
